### PR TITLE
feat: Export API Errors

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
This PR exports some errors we throw inside of `@faststore/api` so starters can catch these errors and respond accordingly. For instance, starters can respond 404 when a `NotFoundError` is thrown inside `@faststore/api`. 

### Starters Deploy Preview
- https://github.com/vtex-sites/nextjs.store/pull/116
- https://github.com/vtex-sites/gatsby.store/pull/108

### CODE REVIEW

File: packages/api/src/platforms/vtex/loaders/sku.ts [hunk 1]
```diff
@@ -39,8 +39,8 @@ export const getSkuLoader = (_: Options, clients: Clients) => {
     const missingSkus = skus.filter((sku) => !sku)
 
     if (missingSkus.length > 0) {
-      throw new Error(
-        `Search API did not return the following skus: ${missingSkus.join(',')}`
+      throw new NotFoundError(
+        `Search API did not found the following skus: ${missingSkus.join(',')}`
```
Comment (eduardoformiga) [inline packages/api/src/platforms/vtex/loaders/sku.ts:43]: [By the time I tested it](https://github.com/vtex/faststore/pull/1220#pullrequestreview-936690157), it was not working as expected.

Maybe it's a good opportunity to improve this logic here. What do you think?

**Review by eduardoformiga — COMMENTED** [review]
Nice! 

In the past, I tested this SKU loader and it's not working as expected.
Maybe it's a good opportunity to change it now.

File: packages/api/src/platforms/vtex/loaders/sku.ts (hunk 1 above)
Comment (tlgimenes) [reply packages/api/src/platforms/vtex/loaders/sku.ts:43]: NICE! Sure!

**Review by eduardoformiga — APPROVED** [review]
Nice! It worked as expected.

Screen Shot 2022-06-13 at 12 54 53

**Pull request merged**